internal/rpc/rpc.go

```diff
@@ -118,12 +120,29 @@ func (rpc *Client) checkSupportedMethods() error {
 	}
 	log.Debug().Msg("eth_getBlockByNumber method supported")
 
-	var getLogsResult interface{}
-	logsErr := rpc.RPCClient.Call(&getLogsResult, "eth_getLogs", map[string]string{"fromBlock": "0x0", "toBlock": "0x0"})
-	if logsErr != nil {
-		return fmt.Errorf("eth_getLogs method not supported: %v", logsErr)
+	if config.Cfg.RPC.BlockReceipts.Enabled {
+		var getBlockReceiptsResult interface{}
+		receiptsErr := rpc.RPCClient.Call(&getBlockReceiptsResult, "eth_getBlockReceipts", "latest")
+		if receiptsErr != nil {
+			log.Warn().Err(receiptsErr).Msg("eth_getBlockReceipts method not supported")
+			return fmt.Errorf("eth_getBlockReceipts method not supported: %v", receiptsErr)
```

Comment:
the idea of returning here is that if `BlockReceipts.Enabled` is is true, but not supported then we don't failover to `eth_getLogs` and stop the execution essentially requiring a user to forcefully set `BlockReceipts.Enabled` to `false`?

Reply:
yes. This is something they explicitly set and this influences the data we get (receipt data) so it would make sense to stop

Reply:
got it, the only small issue here is that we default to it so it makes the default config less OOB-ready now in case RPC doesn't support it

but I it's a double-sided one and I'm ok with leaving it this way, so feel free to resolve if Amine doesn't mind as well

Reply:
Changed default to false